File: truncate_script.py
#!/usr/bin/env python3
import os

filePath = r'D:\madkontrol-app\public\modules\egenkontrol\risikoanalyse.html'

# Read the file
with open(filePath, 'r', encoding='utf-8') as f:
    content = f.read()

# Find the first occurrence of </html>
htmlCloseIndex = content.find('</html>')

if htmlCloseIndex == -1:
    print('ERROR: </html> tag not found')
    exit(1)

# Include the </html> tag (7 characters)
truncateLength = htmlCloseIndex + 7

# Truncate at that position
truncatedContent = content[:truncateLength]

# Write back
with open(filePath, 'w', encoding='utf-8') as f:
    f.write(truncatedContent)

print(f'File truncated successfully!')
print(f'Removed {len(content) - len(truncatedContent)} characters')
print(f'File now ends at character position {truncateLength}')

File: truncate_script.js
const fs = require('fs');
const path = require('path');

const filePath = path.join('D:', 'madkontrol-app', 'public', 'modules', 'egenkontrol', 'risikoanalyse.html');

// Read the file
const content = fs.readFileSync(filePath, 'utf-8');

// Find the first occurrence of </html>
const htmlCloseIndex = content.indexOf('</html>');

if (htmlCloseIndex === -1) {
    console.log('ERROR: </html> tag not found');
    process.exit(1);
}

// Include the </html> tag (7 characters)
const truncateLength = htmlCloseIndex + 7;

// Truncate at that position
const truncatedContent = content.substring(0, truncateLength);

// Write back
fs.writeFileSync(filePath, truncatedContent, 'utf-8');

console.log('File truncated successfully!');
console.log(`Removed ${content.length - truncatedContent.length} characters`);
console.log(`Original size: ${content.length} characters`);
console.log(`New size: ${truncatedContent.length} characters`);
